src/libraries/System.Private.CoreLib/src/System/Numerics/Vector3.cs

```diff
@@ -232,9 +232,7 @@ public static Vector3 Clamp(Vector3 value1, Vector3 min, Vector3 max)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector3 Lerp(Vector3 value1, Vector3 value2, float amount)
         {
-            Vector3 firstInfluence = value1 * (1f - amount);
```

Comment:
The algorithm differs here and will cause different results in some cases due to the order of operations changing.

The difference is that in the case of `v1 + (v2 - v1) * a`, `result != v2 when a == 1`. On the other hand, `(v1 * (1 - a) ) + (v2 * a)` will guarantee it.
The former algorithm is simpler and can be used when `FMA` exists.

I'd guess this was done as a bug fix at some point (I wasn't on the team at the time and I'd have to go digging into TFS to determine the original context), so we should probably preserve the behavior.

Likewise, `Vector2/4` should probably be normalized to do the same, but it is a breaking change and we'd need sign-off (we can do that either in this pr or log an issue and follow up separately).

Reply:
Yes, I noticed that and reverted 🙂

Reply:
I logged https://github.com/dotnet/runtime/issues/35529 to track normalizing all three to the same algorithm